abi/src/main/java/org/web3j/abi/TypeDecoder.java

```diff
@@ -640,24 +651,55 @@ static List arrayToList(Object array) {
 
         try {
             Class<T> cls = Utils.getParameterizedTypeFromArray(typeReference);
-            if (Array.class.isAssignableFrom(cls)) {
-                throw new UnsupportedOperationException(
-                        "Arrays of arrays are not currently supported for external functions, see"
-                                + "http://solidity.readthedocs.io/en/develop/types.html#members");
-            } else {
+            if (StructType.class.isAssignableFrom(cls)) {
                 List<T> elements = new ArrayList<>(length);
```

Comment:
Could this if-else branching be splitted into private functions? It makes the function too long to read.